dask/dataframe/io/hdf.py

```diff
@@ -405,6 +406,12 @@ def read_hdf(pattern, key, start=0, stop=None, columns=None,
         lock = get_scheduler_lock()
 
     key = key if key.startswith('/') else '/' + key
+    # Convert pathlib.Path and objects supporting the fspath protocol to a string
+    if hasattr(pattern, '__fspath__'):
```

Comment:
Should this use the standard `os.fspath()`? Does a Path not have an fspath?

Reply:
fspath was added in 3.6. Once we drop 3.5 we can use it, but for now this is the recommended way to do it.